plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/xpl/ImageName.java

```diff
@@ -0,0 +1,198 @@
+ /** 
+ * Copyright 2014 Roland Huss
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.tools.openshift.internal.ui.wizard.deployimage.xpl;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Copied from https://github.com/rhuss/docker-maven-plugin/blob/6295761106bbe395d90aa7a20c1f314855c0ad23/src/main/java/org/jolokia/docker/maven/util/ImageName.java
+ * 
+ * Helper class for parsing docker repository/image names:
+ *
+ * <ul>
+ *     <li>If the first part before the slash contains a "." or a ":" it is considered to be a registry URL</li>
+ *     <li>A last part starting with with a ":" is considered to be a tag</li>
+ *     <li>The rest is considered the repository name (which might be separated via slashes)</li>
+ * </ul>
+ *
+ * Example of valid names:
+ *
+ * <ul>
+ *     <li>consol/tomcat-8.0</li>
+ *     <li>consol/tomcat-8.0:8.0.9</li>
+ *     <li>docker.consol.de:5000/tomcat-8.0</li>
+ *     <li>docker.consol.de:5000/jolokia/tomcat-8.0:8.0.9</li>
+ * </ul>
+ *
+ * @author roland
+ * @since 22.07.14
+ */
+public class ImageName {
```

Comment:
The openshift rest client has a class called DockerImageURI that already does all this: https://github.com/openshift/openshift-restclient-java/blob/master/src/main/java/com/openshift/restclient/images/DockerImageURI.java